File: 6M3W/ehowlqk/boj_9375.py
T = int(input())
for _ in range(T):
    n = int(input())
    closet = dict()

    for __ in range(n):
        cloth, category = input().split()
        # 이미 등록된 카테고리: 기존 배열에 append
        if closet.get(category):
            closet[category].append(cloth)
        # 새로 등록된 카테고리: 새로운 배열에 담아 등록
        else:
            closet[category] = [cloth]

    answer = 1

    for key in closet:
        # 경우의 수 구하기: <종류 별 의상 개수 + 1>을 모두 곱한다(+1은 착용하지 않는 경우)
        answer *= len(closet[key]) + 1

    print(answer - 1)

File: 6M3W/minjeeki/boj_9375.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

let line_idx = 0;
const case_num = Number(input[line_idx++]);
for (let case_idx = 0; case_idx < case_num; case_idx ++) {
	const n = Number(input[line_idx++]);
	let closet = new Map();
	
	for (let i = 0; i < n; i++) {
		const [wear_name, wear_type] = input[line_idx++].split(" ");
		if (closet.has(wear_type)) {
			closet.get(wear_type).add(wear_name);
		} else {
			closet.set(wear_type, new Set([wear_name]));
		}
	}
		
	// 계산
	let total_case = 1;
	let key_size = closet.size;
	for (const set_value of closet.values()) {
		total_case *= (set_value.size + 1)
	}
	total_case -= 1
	
	// 출력
	console.log(total_case)
}
